### PR TITLE
Add support for SLO burn panel.  

### PR DESCRIPTION
**Pull request by tobio**
Related to https://github.com/elastic/terraform-provider-elasticstack/issues/1736

> [!NOTE]
> ### Add `slo_burn_rate` panel type support to the Kibana dashboard resource
> - Adds a new `slo_burn_rate_config` schema block to the panel schema with required fields `slo_id` and `duration` (validated against `^\d+[mhd]$`) and optional fields `slo_instance_id`, `title`, `description`, `hide_title`, `hide_border`, and `drilldowns`.
> - Implements serialization ([models_slo_burn_rate_panel.go](https://github.com/elastic/terraform-provider-elasticstack/pull/2069/files#diff-3fe517bbc692347f3ac67189a145c0af3f320f180187fb6400a279b05bf2d3e0)) to map TF config to the `slo-burn-rate-embeddable` API schema; drilldowns hardcode `trigger = "on_open_panel_menu"` and `type = "url_drilldown"` — these fields are not exposed to practitioners.
> - Implements deserialization with null-preservation: `slo_instance_id` stays null in state when it was omitted and the API returns the sentinel `"*"`, and optional string/bool fields are only refreshed if they were previously known.
> - Adds plan-time validation that `slo_burn_rate_config` is present when `type = "slo_burn_rate"` and that `config_json` is rejected for this panel type.
> - Covers the feature with unit tests and seven acceptance test scenarios including invalid config, drilldowns, display options, and null-preservation round-trips.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 60894d5.</sup>
> 
>

### CODE REVIEW

File: internal/kibana/dashboard/models_slo_burn_rate_panel_test.go [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+	}
+	return c
+}
+
+func withSloInstanceID(id string) func(*kbapi.SloBurnRateEmbeddable) {
+	return func(c *kbapi.SloBurnRateEmbeddable) { c.SloInstanceId = new(id) }
+}
+
+func withTitle(t string) func(*kbapi.SloBurnRateEmbeddable) {
+	return func(c *kbapi.SloBurnRateEmbeddable) { c.Title = new(t) }
```
[the remark this answers is not in the document]
Comment (tobio) [reply internal/kibana/dashboard/models_slo_burn_rate_panel_test.go:47]: This is valid in go1.26

File: internal/kibana/dashboard/models_slo_burn_rate_panel_test.go [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+}
+
+func withDescription(d string) func(*kbapi.SloBurnRateEmbeddable) {
+	return func(c *kbapi.SloBurnRateEmbeddable) { c.Description = new(d) }
+}
```
[the remark this answers is not in the document]
Comment (tobio) [reply internal/kibana/dashboard/models_slo_burn_rate_panel_test.go:52]: Valid in go1.26
